### ISSUE TITLE
Parsing issue with CTE in Materialized View

### ISSUE THREAD

**Issue by jonyscathe** [sqlfluff/sqlfluff#3092]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

I am getting an unparseable section error on valid pgsql.

Replicated the issue in a simpler case by taking the postgresql [SELECT in WITH example](https://www.postgresql.org/docs/current/queries-with.html), formatting it to pass my sqlfluff linting settings and putting a simple materialized view around it (CREATE AS at the top, replacing the last ";" with a "WITH DATA;" statement:

```
CREATE MATERIALIZED VIEW public.mv_sales
TABLESPACE pg_default
AS

WITH regional_sales AS (
    SELECT
        region
        , SUM(amount) AS total_sales
    FROM orders
    GROUP BY region
)

, top_regions AS (
    SELECT region
    FROM regional_sales
    WHERE total_sales > (SELECT SUM(total_sales) / 10 FROM regional_sales)
)

SELECT
    region
    , product
    , SUM(quantity) AS product_units
    , SUM(amount) AS product_sales
FROM orders
WHERE region IN (SELECT region FROM top_regions)
GROUP BY region, product
WITH DATA;
```

This leads to an error:
```
L:  26 | P:  25 |  PRS | Line 26, Position 25: Found unparsable section: '\nWITH
                       | DATA'
```


Happy to work towards a PR to fix this, but I have absolutely no idea what the issue is at this point so would need to be pointed in the right direction.

### Expected Behaviour

I would expect no parsing issues with the described code.

### Observed Behaviour

Lint output:
```
==== sqlfluff ====
sqlfluff:               0.12.0 python:                 3.10.4
implementation:        cpython verbosity:                   1
dialect:              postgres templater:               jinja

==== readout ====

=== [ path: test.sql ] ===

== [test.sql] FAIL                                                                                                                                                                                                                                                                                                                                                                                     
L:  26 | P:  25 |  PRS | Line 26, Position 25: Found unparsable section: '\nWITH
                       | DATA'
==== summary ====
violations:        1 status:         FAIL
```

Parse output:
```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        create_materialized_view_statement:
[L:  1, P:  1]      |            keyword:                                          'CREATE'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            keyword:                                          'MATERIALIZED'
[L:  1, P: 20]      |            whitespace:                                       ' '
[L:  1, P: 21]      |            keyword:                                          'VIEW'
[L:  1, P: 25]      |            whitespace:                                       ' '
[L:  1, P: 26]      |            table_reference:
[L:  1, P: 26]      |                identifier:                                   'public'
[L:  1, P: 32]      |                dot:                                          '.'
[L:  1, P: 33]      |                identifier:                                   'mv_sales'
[L:  1, P: 41]      |            newline:                                          '\n'
[L:  2, P:  1]      |            keyword:                                          'TABLESPACE'
[L:  2, P: 11]      |            whitespace:                                       ' '
[L:  2, P: 12]      |            tablespace_reference:
[L:  2, P: 12]      |                identifier:                                   'pg_default'
[L:  2, P: 22]      |            newline:                                          '\n'
[L:  3, P:  1]      |            keyword:                                          'AS'
[L:  3, P:  3]      |            newline:                                          '\n'
[L:  4, P:  1]      |            newline:                                          '\n'
[L:  5, P:  1]      |            with_compound_statement:
[L:  5, P:  1]      |                keyword:                                      'WITH'
[L:  5, P:  5]      |                whitespace:                                   ' '
[L:  5, P:  6]      |                [META] indent:
[L:  5, P:  6]      |                common_table_expression:
[L:  5, P:  6]      |                    identifier:                               'regional_sales'
[L:  5, P: 20]      |                    whitespace:                               ' '
[L:  5, P: 21]      |                    keyword:                                  'AS'
[L:  5, P: 23]      |                    whitespace:                               ' '
[L:  5, P: 24]      |                    bracketed:
[L:  5, P: 24]      |                        start_bracket:                        '('
[L:  5, P: 25]      |                        [META] indent:
[L:  5, P: 25]      |                        newline:                              '\n'
[L:  6, P:  1]      |                        whitespace:                           '    '
[L:  6, P:  5]      |                        select_statement:
[L:  6, P:  5]      |                            select_clause:
[L:  6, P:  5]      |                                keyword:                      'SELECT'
[L:  6, P: 11]      |                                [META] indent:
[L:  6, P: 11]      |                                newline:                      '\n'
[L:  7, P:  1]      |                                whitespace:                   '        '
[L:  7, P:  9]      |                                select_clause_element:
[L:  7, P:  9]      |                                    column_reference:
[L:  7, P:  9]      |                                        identifier:           'region'
[L:  7, P: 15]      |                                newline:                      '\n'
[L:  8, P:  1]      |                                whitespace:                   '        '
[L:  8, P:  9]      |                                comma:                        ','
[L:  8, P: 10]      |                                whitespace:                   ' '
[L:  8, P: 11]      |                                select_clause_element:
[L:  8, P: 11]      |                                    function:
[L:  8, P: 11]      |                                        function_name:
[L:  8, P: 11]      |                                            function_name_identifier:  'SUM'
[L:  8, P: 14]      |                                        bracketed:
[L:  8, P: 14]      |                                            start_bracket:    '('
[L:  8, P: 15]      |                                            [META] indent:
[L:  8, P: 15]      |                                            expression:
[L:  8, P: 15]      |                                                column_reference:
[L:  8, P: 15]      |                                                    identifier:  'amount'
[L:  8, P: 21]      |                                            [META] dedent:
[L:  8, P: 21]      |                                            end_bracket:      ')'
[L:  8, P: 22]      |                                    whitespace:               ' '
[L:  8, P: 23]      |                                    alias_expression:
[L:  8, P: 23]      |                                        keyword:              'AS'
[L:  8, P: 25]      |                                        whitespace:           ' '
[L:  8, P: 26]      |                                        identifier:           'total_sales'
[L:  8, P: 37]      |                            newline:                          '\n'
[L:  9, P:  1]      |                            whitespace:                       '    '
[L:  9, P:  5]      |                            [META] dedent:
[L:  9, P:  5]      |                            from_clause:
[L:  9, P:  5]      |                                keyword:                      'FROM'
[L:  9, P:  9]      |                                whitespace:                   ' '
[L:  9, P: 10]      |                                from_expression:
[L:  9, P: 10]      |                                    [META] indent:
[L:  9, P: 10]      |                                    from_expression_element:
[L:  9, P: 10]      |                                        table_expression:
[L:  9, P: 10]      |                                            table_reference:
[L:  9, P: 10]      |                                                identifier:   'orders'
[L:  9, P: 16]      |                                    [META] dedent:
[L:  9, P: 16]      |                            newline:                          '\n'
[L: 10, P:  1]      |                            whitespace:                       '    '
[L: 10, P:  5]      |                            groupby_clause:
[L: 10, P:  5]      |                                keyword:                      'GROUP'
[L: 10, P: 10]      |                                whitespace:                   ' '
[L: 10, P: 11]      |                                keyword:                      'BY'
[L: 10, P: 13]      |                                [META] indent:
[L: 10, P: 13]      |                                whitespace:                   ' '
[L: 10, P: 14]      |                                column_reference:
[L: 10, P: 14]      |                                    identifier:               'region'
[L: 10, P: 20]      |                                [META] dedent:
[L: 10, P: 20]      |                        newline:                              '\n'
[L: 11, P:  1]      |                        [META] dedent:
[L: 11, P:  1]      |                        end_bracket:                          ')'
[L: 11, P:  2]      |                newline:                                      '\n'
[L: 12, P:  1]      |                newline:                                      '\n'
[L: 13, P:  1]      |                comma:                                        ','
[L: 13, P:  2]      |                whitespace:                                   ' '
[L: 13, P:  3]      |                common_table_expression:
[L: 13, P:  3]      |                    identifier:                               'top_regions'
[L: 13, P: 14]      |                    whitespace:                               ' '
[L: 13, P: 15]      |                    keyword:                                  'AS'
[L: 13, P: 17]      |                    whitespace:                               ' '
[L: 13, P: 18]      |                    bracketed:
[L: 13, P: 18]      |                        start_bracket:                        '('
[L: 13, P: 19]      |                        [META] indent:
[L: 13, P: 19]      |                        newline:                              '\n'
[L: 14, P:  1]      |                        whitespace:                           '    '
[L: 14, P:  5]      |                        select_statement:
[L: 14, P:  5]      |                            select_clause:
[L: 14, P:  5]      |                                keyword:                      'SELECT'
[L: 14, P: 11]      |                                [META] indent:
[L: 14, P: 11]      |                                whitespace:                   ' '
[L: 14, P: 12]      |                                select_clause_element:
[L: 14, P: 12]      |                                    column_reference:
[L: 14, P: 12]      |                                        identifier:           'region'
[L: 14, P: 18]      |                            newline:                          '\n'
[L: 15, P:  1]      |                            whitespace:                       '    '
[L: 15, P:  5]      |                            [META] dedent:
[L: 15, P:  5]      |                            from_clause:
[L: 15, P:  5]      |                                keyword:                      'FROM'
[L: 15, P:  9]      |                                whitespace:                   ' '
[L: 15, P: 10]      |                                from_expression:
[L: 15, P: 10]      |                                    [META] indent:
[L: 15, P: 10]      |                                    from_expression_element:
[L: 15, P: 10]      |                                        table_expression:
[L: 15, P: 10]      |                                            table_reference:
[L: 15, P: 10]      |                                                identifier:   'regional_sales'
[L: 15, P: 24]      |                                    [META] dedent:
[L: 15, P: 24]      |                            newline:                          '\n'
[L: 16, P:  1]      |                            whitespace:                       '    '
[L: 16, P:  5]      |                            where_clause:
[L: 16, P:  5]      |                                keyword:                      'WHERE'
[L: 16, P: 10]      |                                [META] indent:
[L: 16, P: 10]      |                                whitespace:                   ' '
[L: 16, P: 11]      |                                expression:
[L: 16, P: 11]      |                                    column_reference:
[L: 16, P: 11]      |                                        identifier:           'total_sales'
[L: 16, P: 22]      |                                    whitespace:               ' '
[L: 16, P: 23]      |                                    comparison_operator:
[L: 16, P: 23]      |                                        raw_comparison_operator:  '>'
[L: 16, P: 24]      |                                    whitespace:               ' '
[L: 16, P: 25]      |                                    bracketed:
[L: 16, P: 25]      |                                        start_bracket:        '('
[L: 16, P: 26]      |                                        [META] indent:
[L: 16, P: 26]      |                                        expression:
[L: 16, P: 26]      |                                            select_statement:
[L: 16, P: 26]      |                                                select_clause:
[L: 16, P: 26]      |                                                    keyword:  'SELECT'
[L: 16, P: 32]      |                                                    [META] indent:
[L: 16, P: 32]      |                                                    whitespace:  ' '
[L: 16, P: 33]      |                                                    select_clause_element:
[L: 16, P: 33]      |                                                        expression:
[L: 16, P: 33]      |                                                            function:
[L: 16, P: 33]      |                                                                function_name:
[L: 16, P: 33]      |                                                                    function_name_identifier:  'SUM'
[L: 16, P: 36]      |                                                                bracketed:
[L: 16, P: 36]      |                                                                    start_bracket:  '('
[L: 16, P: 37]      |                                                                    [META] indent:
[L: 16, P: 37]      |                                                                    expression:
[L: 16, P: 37]      |                                                                        column_reference:
[L: 16, P: 37]      |                                                                            identifier:  'total_sales'
[L: 16, P: 48]      |                                                                    [META] dedent:
[L: 16, P: 48]      |                                                                    end_bracket:  ')'
[L: 16, P: 49]      |                                                            whitespace:  ' '
[L: 16, P: 50]      |                                                            binary_operator:  '/'
[L: 16, P: 51]      |                                                            whitespace:  ' '
[L: 16, P: 52]      |                                                            literal:  '10'
[L: 16, P: 54]      |                                                whitespace:   ' '
[L: 16, P: 55]      |                                                [META] dedent:
[L: 16, P: 55]      |                                                from_clause:
[L: 16, P: 55]      |                                                    keyword:  'FROM'
[L: 16, P: 59]      |                                                    whitespace:  ' '
[L: 16, P: 60]      |                                                    from_expression:
[L: 16, P: 60]      |                                                        [META] indent:
[L: 16, P: 60]      |                                                        from_expression_element:
[L: 16, P: 60]      |                                                            table_expression:
[L: 16, P: 60]      |                                                                table_reference:
[L: 16, P: 60]      |                                                                    identifier:  'regional_sales'
[L: 16, P: 74]      |                                                        [META] dedent:
[L: 16, P: 74]      |                                        [META] dedent:
[L: 16, P: 74]      |                                        end_bracket:          ')'
[L: 16, P: 75]      |                                [META] dedent:
[L: 16, P: 75]      |                        newline:                              '\n'
[L: 17, P:  1]      |                        [META] dedent:
[L: 17, P:  1]      |                        end_bracket:                          ')'
[L: 17, P:  2]      |                newline:                                      '\n'
[L: 18, P:  1]      |                newline:                                      '\n'
[L: 19, P:  1]      |                [META] dedent:
[L: 19, P:  1]      |                select_statement:
[L: 19, P:  1]      |                    select_clause:
[L: 19, P:  1]      |                        keyword:                              'SELECT'
[L: 19, P:  7]      |                        [META] indent:
[L: 19, P:  7]      |                        newline:                              '\n'
[L: 20, P:  1]      |                        whitespace:                           '    '
[L: 20, P:  5]      |                        select_clause_element:
[L: 20, P:  5]      |                            column_reference:
[L: 20, P:  5]      |                                identifier:                   'region'
[L: 20, P: 11]      |                        newline:                              '\n'
[L: 21, P:  1]      |                        whitespace:                           '    '
[L: 21, P:  5]      |                        comma:                                ','
[L: 21, P:  6]      |                        whitespace:                           ' '
[L: 21, P:  7]      |                        select_clause_element:
[L: 21, P:  7]      |                            column_reference:
[L: 21, P:  7]      |                                identifier:                   'product'
[L: 21, P: 14]      |                        newline:                              '\n'
[L: 22, P:  1]      |                        whitespace:                           '    '
[L: 22, P:  5]      |                        comma:                                ','
[L: 22, P:  6]      |                        whitespace:                           ' '
[L: 22, P:  7]      |                        select_clause_element:
[L: 22, P:  7]      |                            function:
[L: 22, P:  7]      |                                function_name:
[L: 22, P:  7]      |                                    function_name_identifier:  'SUM'
[L: 22, P: 10]      |                                bracketed:
[L: 22, P: 10]      |                                    start_bracket:            '('
[L: 22, P: 11]      |                                    [META] indent:
[L: 22, P: 11]      |                                    expression:
[L: 22, P: 11]      |                                        column_reference:
[L: 22, P: 11]      |                                            identifier:       'quantity'
[L: 22, P: 19]      |                                    [META] dedent:
[L: 22, P: 19]      |                                    end_bracket:              ')'
[L: 22, P: 20]      |                            whitespace:                       ' '
[L: 22, P: 21]      |                            alias_expression:
[L: 22, P: 21]      |                                keyword:                      'AS'
[L: 22, P: 23]      |                                whitespace:                   ' '
[L: 22, P: 24]      |                                identifier:                   'product_units'
[L: 22, P: 37]      |                        newline:                              '\n'
[L: 23, P:  1]      |                        whitespace:                           '    '
[L: 23, P:  5]      |                        comma:                                ','
[L: 23, P:  6]      |                        whitespace:                           ' '
[L: 23, P:  7]      |                        select_clause_element:
[L: 23, P:  7]      |                            function:
[L: 23, P:  7]      |                                function_name:
[L: 23, P:  7]      |                                    function_name_identifier:  'SUM'
[L: 23, P: 10]      |                                bracketed:
[L: 23, P: 10]      |                                    start_bracket:            '('
[L: 23, P: 11]      |                                    [META] indent:
[L: 23, P: 11]      |                                    expression:
[L: 23, P: 11]      |                                        column_reference:
[L: 23, P: 11]      |                                            identifier:       'amount'
[L: 23, P: 17]      |                                    [META] dedent:
[L: 23, P: 17]      |                                    end_bracket:              ')'
[L: 23, P: 18]      |                            whitespace:                       ' '
[L: 23, P: 19]      |                            alias_expression:
[L: 23, P: 19]      |                                keyword:                      'AS'
[L: 23, P: 21]      |                                whitespace:                   ' '
[L: 23, P: 22]      |                                identifier:                   'product_sales'
[L: 23, P: 35]      |                    newline:                                  '\n'
[L: 24, P:  1]      |                    [META] dedent:
[L: 24, P:  1]      |                    from_clause:
[L: 24, P:  1]      |                        keyword:                              'FROM'
[L: 24, P:  5]      |                        whitespace:                           ' '
[L: 24, P:  6]      |                        from_expression:
[L: 24, P:  6]      |                            [META] indent:
[L: 24, P:  6]      |                            from_expression_element:
[L: 24, P:  6]      |                                table_expression:
[L: 24, P:  6]      |                                    table_reference:
[L: 24, P:  6]      |                                        identifier:           'orders'
[L: 24, P: 12]      |                            [META] dedent:
[L: 24, P: 12]      |                    newline:                                  '\n'
[L: 25, P:  1]      |                    where_clause:
[L: 25, P:  1]      |                        keyword:                              'WHERE'
[L: 25, P:  6]      |                        [META] indent:
[L: 25, P:  6]      |                        whitespace:                           ' '
[L: 25, P:  7]      |                        expression:
[L: 25, P:  7]      |                            column_reference:
[L: 25, P:  7]      |                                identifier:                   'region'
[L: 25, P: 13]      |                            whitespace:                       ' '
[L: 25, P: 14]      |                            keyword:                          'IN'
[L: 25, P: 16]      |                            whitespace:                       ' '
[L: 25, P: 17]      |                            bracketed:
[L: 25, P: 17]      |                                start_bracket:                '('
[L: 25, P: 18]      |                                [META] indent:
[L: 25, P: 18]      |                                select_statement:
[L: 25, P: 18]      |                                    select_clause:
[L: 25, P: 18]      |                                        keyword:              'SELECT'
[L: 25, P: 24]      |                                        [META] indent:
[L: 25, P: 24]      |                                        whitespace:           ' '
[L: 25, P: 25]      |                                        select_clause_element:
[L: 25, P: 25]      |                                            column_reference:
[L: 25, P: 25]      |                                                identifier:   'region'
[L: 25, P: 31]      |                                    whitespace:               ' '
[L: 25, P: 32]      |                                    [META] dedent:
[L: 25, P: 32]      |                                    from_clause:
[L: 25, P: 32]      |                                        keyword:              'FROM'
[L: 25, P: 36]      |                                        whitespace:           ' '
[L: 25, P: 37]      |                                        from_expression:
[L: 25, P: 37]      |                                            [META] indent:
[L: 25, P: 37]      |                                            from_expression_element:
[L: 25, P: 37]      |                                                table_expression:
[L: 25, P: 37]      |                                                    table_reference:
[L: 25, P: 37]      |                                                        identifier:  'top_regions'
[L: 25, P: 48]      |                                            [META] dedent:
[L: 25, P: 48]      |                                [META] dedent:
[L: 25, P: 48]      |                                end_bracket:                  ')'
[L: 25, P: 49]      |                        [META] dedent:
[L: 25, P: 49]      |                    newline:                                  '\n'
[L: 26, P:  1]      |                    groupby_clause:
[L: 26, P:  1]      |                        keyword:                              'GROUP'
[L: 26, P:  6]      |                        whitespace:                           ' '
[L: 26, P:  7]      |                        keyword:                              'BY'
[L: 26, P:  9]      |                        [META] indent:
[L: 26, P:  9]      |                        whitespace:                           ' '
[L: 26, P: 10]      |                        column_reference:
[L: 26, P: 10]      |                            identifier:                       'region'
[L: 26, P: 16]      |                        comma:                                ','
[L: 26, P: 17]      |                        whitespace:                           ' '
[L: 26, P: 18]      |                        column_reference:
[L: 26, P: 18]      |                            identifier:                       'product'
[L: 26, P: 25]      |                        [META] dedent:
[L: 26, P: 25]      |                unparsable:                                   !! Expected: 'Nothing...'
[L: 26, P: 25]      |                    newline:                                  '\n'
[L: 27, P:  1]      |                    with_data_clause:
[L: 27, P:  1]      |                        keyword:                              'WITH'
[L: 27, P:  5]      |                        whitespace:                           ' '
[L: 27, P:  6]      |                        keyword:                              'DATA'
[L: 27, P: 10]      |    statement_terminator:                                     ';'
[L: 27, P: 11]      |    newline:                                                  '\n'
```


### How to reproduce

Lint the sql from the 'What Happened' section

### Dialect

postgres

### Version

0.12.0
python 3.10.4

### Configuration

[tool.sqlfluff.core]
dialect = 'postgres'
sql_file_exts = '.sql'
exclude_rules = 'L044'

[tool.sqlfluff.indentation]
indented_joins = false
indented_ctes = true
indented_using_on = true
template_blocks_indent = true

[tool.sqlfluff.rules]
tab_space_size = 4
max_line_length = 120
indent_unit = 'space'
comma_style = 'leading'
allow_scalar = true
single_table_references = 'consistent'
unquoted_identifiers_policy = 'all'

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
> Happy to work towards a PR to fix this, but I have absolutely no idea what the issue is at this point so would need to be pointed in the right direction.

The parse tree is useful here to see what's going on. Cutting it down we get this:

```
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        create_materialized_view_statement:
...
[L:  5, P:  1]      |            with_compound_statement:
...
[L: 26, P: 25]      |                unparsable:                                   !! Expected: 'Nothing...'
[L: 26, P: 25]      |                    newline:                                  '\n'
[L: 27, P:  1]      |                    with_data_clause:
[L: 27, P:  1]      |                        keyword:                              'WITH'
[L: 27, P:  5]      |                        whitespace:                           ' '
[L: 27, P:  6]      |                        keyword:                              'DATA'
```

So you can see that, based on the indentation, the parse seems to think the `WITH DATA` bit belongs to the `with_compound_statement`, when it should - it's part of the `create_materialized_view_statement`.

The `with_compound_statement` is here:

https://github.com/sqlfluff/sqlfluff/blob/4bc2916bf87b260744ba77d2d6a6f5ed407a2ba8/src/sqlfluff/dialects/dialect_ansi.py#L2237-L2259

These bit fo the parser has two pieces:

1. a `match_grammar`, which tells the parser exact how much of the SQL to grab, for parsing later. In this case is just has `StartsWith("WITH")`, which means grab anything beginning with `WITH` and then do not stop.
2. an, optional, `parse_grammar` which further says how to parse that SQL chunk you've grabbed in the `match_grammar`.

So the `match_grammar` is over grabbing, and then the `parse_grammar` is not able to parse all that SQL that has been grabbed. Hence the error.

This could be fixed by telling it when to stop by changing it to this:

```
match_grammar: Matchable = StartsWith("WITH", terminator=Ref("WithDataClauseSegment"))
```

However, the `match_grammar` isn't really doing much here, as it's basically just the first part of the `parse_grammar`. A separate `match_grammar` and `parse_grammar` is only really useful if the `match_grammar` helps us filter down (e.g. by having a `StartsWith` AND a`terminator`). If it just has a `StartsWith` and that's the start of the `parse_grammar` anyway, then it's pointless.

So a better option is to just combine these together into one `match_grammar` and drop the separate `parse_grammar` part and that way instead of just `StartsWith` and never ending, it will only match what it can.

```py
    match_grammar: Matchable = Sequence(
        "WITH",
        Ref.keyword("RECURSIVE", optional=True),
        Conditional(Indent, indented_ctes=True),
        Delimited(
            Ref("CTEDefinitionSegment"),
            terminator=Ref.keyword("SELECT"),
        ),
        Conditional(Dedent, indented_ctes=True),
        OneOf(
            Ref("NonWithSelectableGrammar"),
            Ref("NonWithNonSelectableGrammar"),
        ),
    )
```

Doing this allows it to parse.

So with that, and after that adding a test SQL file to `test/fixtures/dialects/postgres` and using the script to generate the equivalent YAML file and you're good to submit a PR. More details on that here: https://github.com/sqlfluff/sqlfluff/wiki/Contributing-Dialect-Changes

**Comment by tunetheweb** [comment]
That’s quite normal for open source projects. You usually don’t get access to the main repo. Instead you need to fork it into your own copy, push to there, and then open a pull request to request we pull the code from your forked copy all the way back into this repo.

I’ve written an extensive git guide here that should explain it all: https://github.com/sqlfluff/sqlfluff/wiki/Git-Guide but happy to answer any further questions on it. It’s a bit confusing at first if you’ve not worked with forks before, but once you get your first PR or two done, it becomes a lot easier.

**Comment by jonyscathe** [comment]
Thanks @tunetheweb. Apologies for my inability to read your well written instructions properly the first time!
Have never used forks before as have only used git on internal projects, but the concept makes perfect sense and seems straightforward enough!
